test/adapters/parsers/json.spec.js

```diff
@@ -56,6 +58,48 @@ describe('parsers/json', function() {
         });
     });
 
+    it('can parse a JSON object with nested arrrays correctly', function() {
+        var json = parsers.getParser('json');
+        var results = [];
+        return json.parseStream(fs.createReadStream(pointWithArrayFile), function(result) {
+            results.push(result);
+        })
+        .then(function() {
+            expect(results.length).equal(1);
+            expect(results).to.deep.equal([[
+                {
+                    'time': '2014-01-01T00:00:00.000Z',
+                    'foo': [
+                        {'fizz1': 'buzz'},
+                        {'fizz2': 'buzz'},
+                        {'fizz3': 'buzz'}
+                    ]
+                }
+            ]]);
+        });
+    });
+
+    it('can parse a JSON array with nested arrrays correctly', function() {
```

Comment:
don't see how that would fail at this point given the checks we have in place using the `path` argument to ignore all nodes that have a path that has more than 1 parent.